src/Entity/Embargo.php

```diff
@@ -43,7 +43,8 @@
  *       "html" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider"
  *     },
  *   },
- *   list_cache_tags = { "node_list", "media_list", "file_list" },
+ *   list_cache_contexts = { "ip.embargo_range", "user" },
+ *   list_cache_tags = { "embargo_list", "embargo_ip_range_list" },
```

Comment:
Have come to realize: Except for `embargo_list` under `list_cache_tags` (which would be the default value, anyhow), none of this should be here, especially `user` under the contexts.

Having the `user` context specified in anything effectively prevents any caching, [as per Drupal's default configuration](https://git.drupalcode.org/project/drupal/-/blob/10.2.x/core/assets/scaffold/files/default.services.yml?ref_type=heads#L128-150)... thinking the inclusion here was intended to be used in some manner to try to deal with user-specific exemptions (and without awareness at the time of poisoning cacheability), but such isn't really relevant to the listing of the embargo entities proper that these keys are meant for. Instead, we're somewhat in the situation with wanting to set cache contexts/tags from our query tagging/altering, [which isn't directly possible](https://www.drupal.org/project/drupal/issues/3028976). There are some other approaches that might make sense to "bubble" cache metadata via render contexts; however, it's not obvious how to nicely account for specific user exemptions. In a sense, it might make sense to add the role-exemption support to allow for "base line" exemptions (for things that would be widely set without completely granting the "bypass" permission to whatever given roles), which should in theory be able to account for more larger swaths of cacheability, allowing caching in the `user.roles` context, instead of the problematically (excessively) granular `user` context.

To deal with specific user exemptions, might get into the situation of:

- producing another context which indicates if a given user is specifically exempt to _any_ embargo
  - if not, can cache in such a context
  - if true, then applying the `user` context to account for the per-user variability of the response/value

Gonna give it something of a look...

Reply:
Have improvements to caching in #40